### PR TITLE
[StructuralMechanicsApplication] Clean up ResidualDisplacementAndOtherDoFCriteria

### PR DESCRIPTION
**Pull request by loumalouomega**
Basically that

### CODE REVIEW

**Comment by loumalouomega** [comment]
I could not find any error on the criteria, I think effectively the shells are broken. Why the tests pass?

**Comment by philbucher** [comment]
Thanks, will check in detail tmr

What makes you think the shells are broken?
Which one are you using?

**Comment by loumalouomega** [comment]
> Thanks, will check in detail tmr
> 
> What makes you think the shells are broken?
> Which one are you using?

It is the thickness, apparently with greater thickness it converges...

File: applications/StructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/residual_displacement_and_other_dof_criteria.h [hunk 1]
```diff
@@ -259,23 +248,22 @@ class ResidualDisplacementAndOtherDoFCriteria : public ConvergenceCriteria< TSpa
...
-        const TSystemVectorType& Dx,
-        const TSystemVectorType& b
-    ) override
+        const TSystemMatrixType& rA,
+        const TSystemVectorType& rDx,
+        const TSystemVectorType& rb
+        ) override
     {
-
+        BaseType::FinalizeSolutionStep(rModelPart, rDofSet, rA, rDx, rb);
```
Comment (philbucher) [inline applications/StructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/residual_displacement_and_other_dof_criteria.h]: can you please remove this method if it only calls the baseclass?

**Review by philbucher — DISMISSED** [review]
minor comment, otherwise thx for cleanup

**Comment by philbucher** [comment]
which shells did you try?

File: applications/StructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/residual_displacement_and_other_dof_criteria.h (hunk 1 above)
Comment (loumalouomega) [reply applications/StructuralMechanicsApplication/custom_strategies/custom_convergencecriterias/residual_displacement_and_other_dof_criteria.h]: I can

**Review by philbucher — APPROVED** [review]
Thx

**Pull request merged**